fv3core/stencils/divergence_damping.py

```diff
@@ -143,16 +146,114 @@ def vorticity_calc(wk, vort, delpc, dt, nord, kstart, nk):
+    return divg_d
+
+
+@gtscript.function
+def divergence_subsequent(
+    rarea_c: FloatField,
+    divg_u: FloatField,
+    divg_v: FloatField,
+    divg_d: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+):
+    from __externals__ import local_ie, local_is, local_je, local_js
+
+    divg_d = corners.fill_corners_bgrid_x(divg_d)
+    with horizontal(region[local_is - 3 : local_ie + 4, local_js - 2 : local_je + 4]):
+        vc = vc_from_divg(divg_d, divg_u)
+    divg_d = corners.fill_corners_bgrid_y(divg_d)
+    with horizontal(region[local_is - 2 : local_ie + 4, local_js - 3 : local_je + 4]):
+        uc = (divg_d[0, 1, 0] - divg_d) * divg_v #uc_from_divg(divg_d, divg_v)
+    vc, uc = corners.fill_corners_dgrid(vc, uc, -1.0)
+    divg_d = update_divg_d(rarea_c, divg_d, uc, vc)
+    return divg_d, uc, vc
+
+@gtscript.function
+def divergence_subsequent1(
+    rarea_c: FloatField,
+    divg_u: FloatField,
+    divg_v: FloatField,
+    divg_d: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+):
+    from __externals__ import local_ie, local_is, local_je, local_js
+    divg_d = corners.fill_corners_bgrid_x(divg_d)
+    with horizontal(region[local_is - 2 : local_ie + 3, local_js - 1 : local_je + 3]):
+        vc = vc_from_divg(divg_d, divg_u)
+    divg_d = corners.fill_corners_bgrid_y(divg_d)
+    with horizontal(region[local_is - 1 : local_ie + 3, local_js - 2 : local_je + 3]):
+        uc = (divg_d[0, 1, 0] - divg_d) * divg_v #uc_from_divg(divg_d, divg_v)                                     
+    vc, uc = corners.fill_corners_dgrid(vc, uc, -1.0)
+    divg_d = update_divg_d(rarea_c, divg_d, uc, vc)
+    return divg_d, uc, vc
+@gtscript.function
+def divergence_at_nord(
+    rarea_c: FloatField,
+    divg_u: FloatField,
+    divg_v: FloatField,
+    divg_d: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+):
+    from __externals__ import local_ie, local_is, local_je, local_js
+    with horizontal(region[local_is - 1 : local_ie + 2, local_js : local_je + 2]):
+        vc = vc_from_divg(divg_d, divg_u)
+    with horizontal(region[local_is : local_ie + 2, local_js - 1 : local_je + 2]):
+        uc = uc_from_divg(divg_d, divg_v)
+    divg_d = update_divg_d(rarea_c, divg_d, uc, vc)
+    return divg_d, uc, vc
+
+
+@gtstencil()
+def damping_nonzero_nord(
+    rarea_c: FloatField,
+    divg_u: FloatField,
+    divg_v: FloatField,
+    divg_d: FloatField,
+    uc: FloatField,
+    vc: FloatField,
+    delpc: FloatField,
+):
+
+    with computation(PARALLEL), interval(...):
+        delpc = divg_d
+        # TODO, can we call the same function 3 times, let gt4py do the extent analysis?
+        # currently does not work because corner calculations need entire array,
+        # and vc/uc need offsets
+        # Another TODO: once there is an idea of 'repetition', we need to loop
+        # namelist.nord times, not always 3 times, that is just what we have
+        # it set to. nord = 0, 1, or 2 are possible options
```

Comment:
I could, though wouldn't be able to test it without making more data, unlike delnflux which actually does get called with different 'nords'. This approach though would make it more complete and not require a new gt4py feature

Reply:
I was mostly kidding about the unrolled implementation since I did that in delnflux. Once we have repetition both delnflux and this should probably use it.

In the meantime, though, if this implementation is only for nord=2, would it make sense to say that in the names or docstrings somewhere? Just so that it's clear that we may have to add other nord modes when we refactor?